lib/mp4/transmuxer.js

```diff
@@ -976,6 +978,7 @@ CoalesceStream.prototype.flush = function(flushSource) {
     caption.startTime /= 90e3;
     caption.endTime = (caption.endPts - timelineStartPts);
     caption.endTime /= 90e3;
+    event.captionTracks[caption.track] = true;
```

Comment:
You will have to make these changes in the `CoalesceStream` for the FLV transmuxer as well https://github.com/videojs/mux.js/blob/master/lib/flv/coalesce-stream.js#L112